recipe_scrapers/coopse.py

```diff
@@ -0,0 +1,110 @@
+# mypy: allow-untyped-defs
+import json
+import re
+
+import requests
+
+from recipe_scrapers._grouping_utils import IngredientGroup
+
+from ._abstract import AbstractScraper
+
+SCRIPT_PATTERN = re.compile('"recipeId":"(?P<recipeId>[0-9]+)"')
+
+
+class CoopSE(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+        scripts = self.soup.find_all("script")
+        recipe_id = None
+        for script in scripts:
+            matches = SCRIPT_PATTERN.search(str(script.string))
+            if matches:
+                data = matches.groupdict()
+                recipe_id = data["recipeId"]
+        data_url = f"https://proxy.api.coop.se/external/recipe/recipes/{recipe_id}?api-version=v1"
+        self.page_data = requests.get(
+            data_url,
+            proxies=proxies,
+            timeout=timeout,
+        ).content
+        self.data = json.loads(self.page_data)
+
+    @classmethod
+    def host(cls):
+        return "coop.se"
+
+    def author(self):
+        return self.data["recipeCreator"]
```

Comment:
In cases where an author isn't attributed on recipe pages, I think we should generally credit the site owner / publisher -- basically it's to help people figure out where they retrieve a recipe from, if they later want to go back and find out.

Oddly `recipeCreator` seems intended for exactly authorship details, and yet is empty.

```suggestion
    def author(self):
        return "Coop Sverige"
```

Reply:
Who knows, maybe they intend to use it in the future..

Reply:
Hmm.  Good point.  Perhaps we could make this further adjustment, to provide the static credit as a fallback:

```python
return self.data["recipeCreator"] or "Coop Sverige"
```

It's difficult to say for certain though, without knowing how the value might appear in future.  What do you think?